### PR TITLE
Add unit tests for WC_Payment_Token_CashApp and WC_Payment_Token_Link

### PR DESCRIPTION
**Pull request by noruzzamans**
## Changes proposed in this Pull Request:

Add dedicated unit test coverage for the `WC_Payment_Token_CashApp` and `WC_Payment_Token_Link` classes.

Sister payment tokens under `includes/payment-tokens/` (ACH, ACSS, Amazon Pay, BACS Debit, BECS Debit, CC, Klarna, and SEPA) each have a corresponding test suite in `tests/phpunit/payment-tokens/`. This PR adds test coverage for the remaining untested token classes:

- `tests/phpunit/payment-tokens/class-wc-payment-token-cashapp-test.php`:
  - Token type verification (`WC_Stripe_Payment_Methods::CASHAPP_PAY`)
  - Localized display name formatting with and without `$cashtag`
  - Token validation constraints (`validate()`) via data provider
  - Getter and setter pairs (`cashtag`) via data provider
  - Payment method equality comparisons (`is_equal_payment_method()`) covering type and cashtag matching, mismatched type, mismatched cashtag, and empty cashtag scenarios

- `tests/phpunit/payment-tokens/class-wc-payment-token-link-test.php`:
  - Token type verification (`WC_Stripe_Payment_Methods::LINK`)
  - Display name formatting with and without customer email
  - Token validation constraints (`validate()`) via data provider
  - Getter and setter pairs (`email`) via data provider
  - Payment method equality comparisons (`is_equal_payment_method()`) covering type and email matching, mismatched type, mismatched email, and empty email scenarios

## Testing instructions

Verify static analysis, linting, and automated tests:

1. Run PHPCS:
```bash
./vendor/bin/phpcs --standard=phpcs.xml.dist tests/phpunit/payment-tokens/class-wc-payment-token-cashapp-test.php tests/phpunit/payment-tokens/class-wc-payment-token-link-test.php
```

2. Run PHPStan:
```bash
vendor/bin/phpstan analyse --configuration=phpstan.neon --memory-limit=2G
```

3. Run the PHPUnit tests in the test environment:
```bash
npm run test:php -- --filter="WC_Payment_Token_CashApp_Test|WC_Payment_Token_Link_Test"
```

---

- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### Changelog entry

- [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment
Test-only addition adding unit test coverage for `WC_Payment_Token_CashApp` and `WC_Payment_Token_Link`.

</details>

### CODE REVIEW

File: tests/phpunit/payment-tokens/class-wc-payment-token-cashapp-test.php [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	 * Data provider for `test_is_equal_payment_method()`.
+	 *
+	 * @return array
+	 */
+	public function provide_test_is_equal_payment_method(): array {
+		return [
+			'type and cashtag match' => [ '$cashuser', WC_Stripe_Payment_Methods::CASHAPP_PAY, '$cashuser', true, 'is_equal_payment_method() should return true when type and cashtag match.' ],
+			'mismatched type'        => [ '$cashuser', 'card', '$cashuser', false, 'is_equal_payment_method() should return false when the type is not cashapp.' ],
+			'mismatched cashtag'     => [ '$cashuser', WC_Stripe_Payment_Methods::CASHAPP_PAY, '$otheruser', false, 'is_equal_payment_method() should return false when the cashtag does not match.' ],
+			'empty token cashtag'    => [ '', WC_Stripe_Payment_Methods::CASHAPP_PAY, '', true, 'is_equal_payment_method() should return true when both cashtags are empty.' ],
```
[the remark this answers is not in the document]
Comment (noruzzamans) [reply tests/phpunit/payment-tokens/class-wc-payment-token-cashapp-test.php]: Thanks for the feedback! Updated in commit `a49d1e54` with dedicated test cases covering incomplete payment method data where `cashapp` is missing and where `cashtag` is missing in `cashapp` (and also mirrored the same edge-case coverage for Link tokens as well).

**Review by daledupreez — APPROVED** [review]
Thanks for the contribution, @noruzzamans! Much appreciated!

These tests look good to me, and we'll get them merged shortly.
